libraries/botframework-schema/src/index.ts

```diff
@@ -17,939 +19,1122 @@ export * from './teams';
+    /**
+     * List of phrases and references that speech and language priming systems should listen for
+     */
+    listenFor: string[];
+    /**
+     * The collection of text fragments to highlight when the activity contains a ReplyToId value.
+     */
+    textHighlights?: TextHighlight[];
+    /**
+     * An optional programmatic action accompanying this request
+     */
+    semanticAction?: SemanticAction;
+}
+
+export const assertActivity: Assertion<Activity> = (value, path) => {
+    assert.unsafe.castObjectAs<Activity>(value, path);
+    assert.string(value.type, path.concat('type'));
+    assert.maybeString(value.id, path.concat('id'));
+    assert.maybeDate(value.timestamp, path.concat('timestamp'));
+    assert.maybeDate(value.localTimestamp, path.concat('localTimestamp'));
+    assert.string(value.localTimezone, path.concat('localTimezone'));
+    assert.string(value.callerId, path.concat('calledId'));
+    assert.string(value.serviceUrl, path.concat('serviceUrl'));
+    assert.string(value.channelId, path.concat('channelId'));
+    assertChannelAccount(value.from, path.concat('from'));
+    assertConversationAccount(value.conversation, path.concat('conversation'));
+    assertChannelAccount(value.recipient, path.concat('recipient'));
+    assert.maybeString(value.textFormat, path.concat('textFormat'));
+    assert.maybeString(value.attachmentLayout, path.concat('attachmentLayout'));
+    assertChannelAccountArray(value.membersAdded, path.concat('membersAdded'));
+    assertChannelAccountArray(value.membersRemoved, path.concat('membersRemoved'));
+    assertMessageReactionArray(value.reactionsAdded, path.concat('reactionsAdded'));
+    assertMessageReactionArray(value.reactionsRemoved, path.concat('reactionsRemoved'));
+    assert.maybeString(value.topicName, path.concat('topicName'));
+    assert.maybeBoolean(value.historyDisclosed, path.concat('historyDisclosed'));
+    assert.maybeString(value.locale, path.concat('locale'));
+    assert.string(value.text, path.concat('text'));
+    assert.maybeString(value.speak, path.concat('speak'));
+    assert.maybeString(value.inputHint, path.concat('inputHint'));
+    assert.maybeString(value.summary, path.concat('summary'));
+    assertSuggestedActions(value.suggestedActions, path.concat('suggestedActions'));
+    assertMaybeAttachmentArray(value.attachments, path.concat('attachments'));
+    assertMaybeEntityArray(value.entities, path.concat('entities'));
+    assert.maybeAny(value.channelData, path.concat('channelData'));
+    assert.maybeString(value.action, path.concat('action'));
+    assert.maybeString(value.replyToId, path.concat('replyToId'));
+    assert.string(value.label, path.concat('label'));
+    assert.string(value.valueType, path.concat('valueType'));
+    assert.maybeAny(value.value, path.concat('value'));
+    assert.maybeString(value.name, path.concat('name'));
+    assertMaybeConversationReference(value.relatesTo, path.concat('relatesTo'));
```

Comment:
You may want to double check with John, Gabo, and/or Eric on this assertion. AFAIK we normally construct the conversationReference from other data found on the Activity. 

`Activity.relatesTo` is only populated in Skills scenarios.

Reply:
That's permissible (maybe implies the value can be null or undefined), this code just matches the interface definition.